### PR TITLE
Use secure sockets if connecting over https

### PR DESCRIPTION
**Pull request by glynhudson**

### CODE REVIEW

File: src/html/view_models/OpenEvseWiFiViewModel.js [hunk 1]
```diff
@@ -16,7 +18,11 @@ function OpenEvseWiFiViewModel(baseHost, basePort)
     return endpoint;
   });
   self.wsEndpoint = ko.pureComputed(function () {
-    var endpoint = "ws://" + self.baseHost();
+    if("https:" === self.baseProtocol()){
+      var endpoint = "wss://" + self.baseHost();
+    } else {
```
Comment (jeremypoulter) [inline src/html/view_models/OpenEvseWiFiViewModel.js]: Don't need the else

File: src/html/view_models/OpenEvseWiFiViewModel.js [hunk 2]
```diff
@@ -16,7 +18,11 @@ function OpenEvseWiFiViewModel(baseHost, basePort)
     return endpoint;
   });
   self.wsEndpoint = ko.pureComputed(function () {
-    var endpoint = "ws://" + self.baseHost();
+    if("https:" === self.baseProtocol()){
+      var endpoint = "wss://" + self.baseHost();
```
Comment (jeremypoulter) [inline src/html/view_models/OpenEvseWiFiViewModel.js]: Remove the var so the default is over written

**Comment by glynhudson** [comment]
Thanks for the suggestions. I've made the changes suggested, see my latest commit to this branch: https://github.com/OpenEVSE/ESP8266_WiFi_v2.x/pull/158/commits/855fa1d896c4d592dd66c1f23fdcace36b007a8e. 

Shall I merge this PR now? Or do you want me to hold off merging until you have pushed some changes with regard to testing you mentioned above?

**Comment by jeremypoulter** [comment]
Go ahead and merge, I will raise a new PR. Thinking about it the changes are not really part of this CR, just to make it easier to setup and test.

**Pull request merged**